damnit/ctxsupport/ctxrunner.py

```diff
@@ -594,6 +598,98 @@ def save_hdf5(self, hdf5_path, reduced_only=False):
         if os.stat(hdf5_path).st_uid == os.getuid():
             os.chmod(hdf5_path, 0o666)
 
+
+@contextmanager
+def filesystem(host='localhost'):
+    """Mount remote proposal data with sshfs
+
+    mount `/gpfs/exfel/exp/` from `host` and patch
+    `extra_data.read_machinery.DATA_ROOT_DIR` to use it instead of the local one.
+    Opening a file or a run directory with extra_data will open the remote data.
+    """
+    if host == 'localhost':
+        yield
+        return
+
+    with TemporaryDirectory() as td:
+        try:
+            mount_command = [
+                "sshfs", f"{host}:{extra_data.read_machinery.DATA_ROOT_DIR}", str(td),
+                # deactivate password prompt to fail if we don't have a valid ssh key
+                "-o", "ssh_command='ssh -o PasswordAuthentication=no'"
+            ]
+            run(mount_command, check=True, shell=True)
+
+            with patch("extra_data.read_machinery.DATA_ROOT_DIR", td):
```

Comment:
Should we make this a parameter for `open_run()` or have some other way of passing it in? It's OK for now, but this is relying on implementation details of extra_data.

Reply:
I agree if we use this in the long run, but considering this is only a temporary solution (look like we may get slurm on the online cluster :crossed_fingers: ) I'd say it's better to not add extra arguments to extra_data